src/datepicker/datepicker.ts

```diff
@@ -271,16 +290,103 @@ export class NgbDatepicker implements OnChanges,
   onNavigateEvent(event: NavigationEvent) {
     switch (event) {
       case NavigationEvent.PREV:
-        this._setViewWithinLimits(this._calendar.getPrev(this.months[0].firstDate, 'm'));
+        this._setRelativeFocusedDate('m', -1);
         break;
       case NavigationEvent.NEXT:
-        this._setViewWithinLimits(this._calendar.getNext(this.months[0].firstDate, 'm'));
+        this._setRelativeFocusedDate('m', 1);
         break;
     }
 
     this._updateData();
   }
 
+  @HostListener('keydown', ['$event'])
+  onKeyDown(event: KeyboardEvent) {
+    let focusedDate = this.focusedDate;
+    if (!focusedDate) {
+      return;
+    }
+    switch (event.keyCode) {
+      case 33 /* page up */:
```

Comment:
In `rating` and  `typeahead` we have an `enum Keys` that contains key codes (and ideally should be shared), so switch reads like:  `case Key.PageUp:`

Not sure which one I prefer though...